speechcolab/datasets/gigaspeech.py

```
@@ -119,6 +120,17 @@ def download_and_process_object_from_release(self, remote_md5, obj):
             with open(local_obj, 'wb') as f:
                 f.write(data)
 
+            # Check md5 of the written file
+            with open(local_obj, 'rb') as f:
+                data = f.read()
+                local_md5 = hashlib.md5(data).hexdigest()
+                if local_md5 == remote_md5:
+                    print(f'Successfully verified md5 for {local_obj}')
+                    need_download = False
+                else:
+                    print(f'$local_version expects md5=$md5, got $local_md5, try downloading')
+                    retry_count -= 1
```

Should we let it fail if retry_count goes down to 0?